assets/js/background.js

```diff
@@ -178,6 +185,103 @@ function handleOnUpdated(tabId, changeInfo, tab) {
 }
 
 /**
+ * Message listenener for context requests
+ */
+var findContext = function findContext(request, sender, sendResponse) {
+  chrome.tabs.query({
+    active: true,
+    currentWindow: true
+  }, function(tabs) {
+    var tab = tabs[0];
+    var site = getSiteFromTab(config.supportedSites, tab);
+    // Unsupported website, skip.
+    if(!site) {
+      console.warn('Unsupported');
+      return;
+    }
+    // Retrieve context from tab
+    detectContextWithRetry(tab, site, 2, 1000, function(err, context) {
+      if(err) {
+        console.warn(err);
+        return;
+      }
+      sendResponse({
+        site: site,
+        context: blacklist.filterQuery(context)
+      });
+      return;
+    });
+  });
+  return true;
+};
+
+
+/**
+ * Message listenener for search requests
+ */
+var getResults = function getResults(request, sender, sendResponse) {
+  var context = request.context;
+  if(!context.length) {
+    return;
+  }
+
+  var query = generateQuery(context);
+  var payload = {
+    query: query,
+    check: request.check
+  };
+
+  async.waterfall([
+    function loadSettings(cb) {
+      if(config.loaded) {
+        return cb(null);
+      }
+      config.loadUserSettings(cb);
+    },
+    function search(cb) {
+      getDocuments(query, cb);
+    },
+    function respond(documents, count, cb) {
+      payload.count = count;
+      payload.documents = documents;
+      sendResponse(payload);
+      cb(null);
+    }
+  ], function(err) {
+    if(err) {
+      console.warn(err);
+    }
+  });
+
+  return true;
+};
+
+/**
+ * Message listenener for toggle context requests
+ */
+var toggleContextItem = function toggleContextItem(request, sender, sendResponse) {
+  request.context.some(function(item, index, context) {
+    if(item.name === request.name) {
+      var newState = !item.active;
+      context[index].active = newState;
+      if(config.blacklist[item.name.toLowerCase()] && newState) {
+        delete config.blacklist[item.name.toLowerCase()];
+      }
+      else if(!newState) {
+        config.blacklist[item.name.toLowerCase()] = true;
+      }
+      return true;
+    }
+    return false;
+  });
+
+  chrome.storage.sync.set({blacklist: config.blacklist});
```

Comment:
Yes, why not ? That was done this way I think (@merlinND did that) to avoid doing too many calls to the Storage API, but the performance gain might be negligible.